### DISCUSSION TITLE
feat: spin up contracts node with pre-defined contracts

### DISCUSSION THREAD

**Discussion by brunopgalvao** [r0gue-io/pop-cli#303] [Feature Request]
When doing smart contract development, it is often the case that one needs more than one smart contract for a proper development environment.

As a smart contract developer, I would like to be able to spin up a contracts node with specific smart contracts uploaded.

This request came from Pablo Blanco:

> I created a bash script that compiles all the contracts from different folders and runs the node for example:
```sh
case $accion in
cargo +stable contract build --manifest-path contracts/psp22/Cargo.toml;;
cargo +stable contract build --manifest-path contracts/psp34_bis/V1/Cargo.toml;;
cargo +stable contract build --manifest-path contracts/psp34_bis/V2/Cargo.toml;;
cargo +stable contract build --manifest-path contracts/reputationSystem/Cargo.toml --skip-wasm-validation;;
cargo +stable contract build --manifest-path contracts/reputationSystem/Cargo.toml
cargo +stable contract build --manifest-path contracts/psp34_bis/V1/Cargo.toml
cargo +stable contract build --manifest-path contracts/psp22/Cargo.toml;;
esac
echo "Not an option.";;
esac
```

**Comment by alexbcasas** [comment]
Interesting feature. 
I have to try it first but I think this can work right now with pop. We have the feature to run a script upon network initialization: https://learn.onpop.io/v/appchains/guides/running-a-post-startup-command

You can have a basic script ready to run:
```
pop new contract my_contract -c psp -t PSP22
pop up contract --path ./my_contract --url your-chain-url
```
And execute it after:
```
pop up parachain -f network.toml --cmd ./script
```
The only thing you'll need to know in advance is the url of your chain, but you can specify the port in the `network.toml`.
